pkg/vgmanager/vgmanager_controller.go

```diff
@@ -243,3 +298,71 @@ func ToleratesTaints(tolerations []corev1.Toleration, taints []corev1.Taint) boo
 	}
 	return true
 }
+
+func setStatus(status *lvmv1alpha1.VGStatus, instance *lvmv1alpha1.LVMVolumeGroupNodeStatus) {
+	found := false
+
+	vgStatuses := instance.Spec.LVMVGStatus
+	for _, vgStatus := range vgStatuses {
+		if vgStatus.Name == status.Name {
+			found = true
+			vgStatus.Status = status.Status
+			vgStatus.Reason = status.Reason
+			break
+		}
+	}
+
+	if !found {
+		newStatus := &lvmv1alpha1.VGStatus{
+			Name:   status.Name,
+			Status: status.Status,
+			Reason: status.Reason,
+		}
+		vgStatuses = append(vgStatuses, *newStatus)
+		instance.Spec.LVMVGStatus = vgStatuses
+	}
+}
+
+func (r *VGReconciler) updateStatus(ctx context.Context, status *lvmv1alpha1.VGStatus, instance *lvmv1alpha1.LVMVolumeGroup) error {
+
+	vgNodeStatus := r.getNewNodeStatus(status)
+
+	nodeStatus := &lvmv1alpha1.LVMVolumeGroupNodeStatus{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      r.NodeName,
+			Namespace: r.Namespace,
+		},
+	}
+
+	result, err := ctrl.CreateOrUpdate(ctx, r.Client, nodeStatus, func() error {
+		if nodeStatus.CreationTimestamp.IsZero() {
+			vgNodeStatus.DeepCopyInto(nodeStatus)
+			return nil
+		}
```

Comment:
- i think `setStatus` already handles this condition by creating a new status when one is not found?

Reply:
This is working on the existing CR in the system so we need to update that object.